app/javascript/components/dropdowns/Dropdown.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { useState, useEffect, useRef } from 'react'
+import { usePanel } from '../../hooks/use-panel'
+
+type MenuButton = {
+  label: string
+  id: string
+  className: string
+  html: string
+}
+
+type MenuItem = {
+  html: string
+  className: string
+}
+
+export const Dropdown = ({
+  menuButton,
+  menuItems,
+}: {
+  menuButton: MenuButton
+  menuItems: MenuItem[]
+}): JSX.Element => {
+  const { open, setOpen, buttonRef, panelRef, styles, attributes } = usePanel()
+  const [focusIndex, setFocusIndex] = useState<number | null>(null)
+  const menuItemElementsRef = useRef<HTMLLIElement[]>([])
+
+  const handleButtonKeyDown = (e: React.KeyboardEvent) => {
+    switch (e.key) {
+      case 'ArrowDown':
+        e.preventDefault()
+        setOpen(true)
+        setFocusIndex(0)
+
+        break
+      case 'ArrowUp':
+        e.preventDefault()
+        setOpen(true)
+        setFocusIndex(menuItems.length - 1)
+
+        break
+    }
+  }
+
+  const handleItemKeyDown = (e: React.KeyboardEvent, index: number) => {
+    switch (e.key) {
+      case 'ArrowDown':
+        e.preventDefault()
+        setFocusIndex((index + menuItems.length + 1) % menuItems.length)
+
+        break
+      case 'ArrowUp':
+        e.preventDefault()
+        setFocusIndex((index + menuItems.length - 1) % menuItems.length)
+
+        break
+      case 'Tab':
+        setOpen(false)
+
+        break
+      case 'Escape':
+        e.preventDefault()
+        setOpen(false)
+        setFocusIndex(null)
+
+        break
+      case ' ':
+      case 'Enter': {
+        e.preventDefault()
+
+        setOpen(false)
+
+        const link = menuItemElementsRef.current[index]?.querySelector('a')
+        link?.click()
+
+        break
+      }
+    }
+  }
+
+  const handleMenuItemMount = (
+    instance: HTMLLIElement | null,
+    index: number
+  ) => {
+    if (!instance) {
+      return
+    }
+
+    menuItemElementsRef.current[index] = instance
+  }
+
+  useEffect(() => {
+    if (focusIndex === null) {
+      buttonRef.current?.focus()
+
+      return
+    }
+
+    menuItemElementsRef.current[focusIndex].focus()
+  }, [open, focusIndex, buttonRef])
+
+  return (
+    <React.Fragment>
+      <button
+        className={`${menuButton.className}`}
+        aria-controls={`${menuButton.id}-dropdown`}
+        aria-haspopup
+        aria-label={menuButton.label}
+        aria-expanded={open ? true : undefined}
+        dangerouslySetInnerHTML={{ __html: menuButton.html }}
+        ref={buttonRef}
+        onKeyDown={handleButtonKeyDown}
+        onClick={() => {
+          setOpen(!open)
+        }}
+      />
+      <div ref={panelRef} style={styles.popper} {...attributes.popper}>
+        <ul
+          className={`${menuButton.className}-dropdown`}
+          id={`${menuButton.id}-dropdown`}
```

Comment:
When setting it, you can have the variable as `dropdownId` rather than `menuButton.id` as we never use the `menuButton.id` without `-dropdown` now.